### PR TITLE
Fix type hints

### PR DESCRIPTION
**Pull request by stchr**
This fixes some type hints for better code analysis.

### CODE REVIEW

File: include/tcpdf_fonts.php [hunk 1]
```diff
@@ -1787,7 +1787,7 @@ public static function UTF8ArrayToUniArray($ta, $isunicode=true) {
 
 	/**
 	 * Extract a slice of the $strarr array and return it as string.
-	 * @param string $strarr The input array of characters.
+	 * @param array $strarr The input array of characters.
```
Comment (williamdes) [inline include/tcpdf_fonts.php:1790]:
```suggestion
	 * @param string[] $strarr The input array of characters.
```
Would it match better ?

File: tcpdf_barcodes_1d.php [hunk 2]
```diff
@@ -243,7 +243,6 @@ public function getBarcodePngData($w=2, $h=30, $color=array(0,0,0)) {
...
- 	 * @return array barcode array
```
Comment (williamdes) [inline tcpdf_barcodes_1d.php:246]: `@return void ?`

File: tcpdf_barcodes_2d.php [hunk 3]
```diff
@@ -247,7 +247,6 @@ public function getBarcodePngData($w=3, $h=3, $color=array(0,0,0)) {
...
 	 * @param string $code code to print
  	 * @param string $type type of barcode: <ul><li>DATAMATRIX : Datamatrix (ISO/IEC 16022)</li><li>PDF417 : PDF417 (ISO/IEC 15438:2006)</li><li>PDF417,a,e,t,s,f,o0,o1,o2,o3,o4,o5,o6 : PDF417 with parameters: a = aspect ratio (width/height); e = error correction level (0-8); t = total number of macro segments; s = macro segment index (0-99998); f = file ID; o0 = File Name (text); o1 = Segment Count (numeric); o2 = Time Stamp (numeric); o3 = Sender (text); o4 = Addressee (text); o5 = File Size (numeric); o6 = Checksum (numeric). NOTES: Parameters t, s and f are required for a Macro Control Block, all other parametrs are optional. To use a comma character ',' on text options, replace it with the character 255: "\xff".</li><li>QRCODE : QRcode Low error correction</li><li>QRCODE,L : QRcode Low error correction</li><li>QRCODE,M : QRcode Medium error correction</li><li>QRCODE,Q : QRcode Better error correction</li><li>QRCODE,H : QR-CODE Best error correction</li><li>RAW: raw mode - comma-separad list of array rows</li><li>RAW2: raw mode - array rows are surrounded by square parenthesis.</li><li>TEST : Test matrix</li></ul>
- 	 * @return array
```
Comment (williamdes) [inline tcpdf_barcodes_2d.php:250]: `@return void`

**Review by williamdes — APPROVED** [review]
Very nice work !
I added some comments

**Comment by stchr** [comment]
@williamdes Thanks for the review, all suggested changes applied 💪

File: include/tcpdf_fonts.php [hunk 4]
```diff
@@ -1810,7 +1810,7 @@ public static function UTF8ArrSubString($strarr, $start='', $end='', $unicode=tr
 
 	/**
 	 * Extract a slice of the $uniarr array and return it as string.
-	 * @param string $uniarr The input array of characters.
+	 * @param array $uniarr The input array of characters.
```
Comment (williamdes) [inline include/tcpdf_fonts.php:1813]:
```suggestion
	 * @param string[] $uniarr The input array of characters.
```
This one too ;)

File: tcpdf.php [hunk 5]
```diff
@@ -4103,7 +4103,7 @@ public function GetStringWidth($s, $fontname='', $fontstyle='', $fontsize=0, $ge
 
 	/**
 	 * Returns the string length of an array of chars in user unit or an array of characters widths. A font must be selected.<br>
-	 * @param string $sa The array of chars whose total length is to be computed
+	 * @param array $sa The array of chars whose total length is to be computed
```
Comment (williamdes) [inline tcpdf.php:4106]: Maybe copy return of `$sa = TCPDF_FONTS::UTF8ArrToLatin1Arr($sa);` ? since the function seems to do a copy

Comment (stchr) [reply tcpdf.php:4106]: @williamdes not in every case. See `@param boolean $getarray if true returns an array of characters widths, if false returns the total length.`
So the return should be `@return float[]|float` imho.

Comment (stchr) [reply tcpdf.php:4106]: I've changed this in 6e988d94de25b1e92c0d26adffe27a87dbc497d7

**Review by williamdes — APPROVED** [review]
Awesome

**Review by nicolaasuni — APPROVED** [review]

**Pull request merged**